diff --git a/plugins/redmine_tags/README.md b/plugins/redmine_tags/README.md
index 73b41ae..7effa94 100644
--- a/plugins/redmine_tags/README.md
+++ b/plugins/redmine_tags/README.md
@@ -35,6 +35,8 @@ Inspired by original `redmine_tags` of Eric Davis.
     ```
 
 3. Restart your Redmine web server.
+4. Assign edit permissions to roles.
+5. Enable Tags per project.
 
 
 ## Running tests
diff --git a/plugins/redmine_tags/app/views/issues/_tags.html.erb b/plugins/redmine_tags/app/views/issues/_tags.html.erb
index 3800666..8dfd0cb 100755
--- a/plugins/redmine_tags/app/views/issues/_tags.html.erb
+++ b/plugins/redmine_tags/app/views/issues/_tags.html.erb
@@ -1,3 +1,4 @@
+<% if @project.module_enabled?(:tags) %>
 <% unless issue.tag_list.empty? %>
   <div class="tags attribute">
     <div class="label">
@@ -8,3 +9,4 @@
     </div>
   </div>
 <% end %>
+<% end %>
diff --git a/plugins/redmine_tags/app/views/issues/_tags_form.html.erb b/plugins/redmine_tags/app/views/issues/_tags_form.html.erb
index 3ef2288..b9bff0f 100755
--- a/plugins/redmine_tags/app/views/issues/_tags_form.html.erb
+++ b/plugins/redmine_tags/app/views/issues/_tags_form.html.erb
@@ -1,3 +1,4 @@
+<% if @project.module_enabled?(:tags) && User.current.allowed_to?(:issue_edit_tags, @project) %>
 <p><%= form.text_field :tag_list, :size => 60, :value => issue.tag_list.to_s %></p>
 
 <%= javascript_tag "$('#issue_tag_list').tagit({
@@ -17,3 +18,4 @@
   removeConfirmation: true,
   showAutocompleteOnFocus: true,
 });" %>
+<% end %>
\ No newline at end of file
diff --git a/plugins/redmine_tags/app/views/issues/_tags_sidebar.html.erb b/plugins/redmine_tags/app/views/issues/_tags_sidebar.html.erb
index 7377164..d51b6a3 100644
--- a/plugins/redmine_tags/app/views/issues/_tags_sidebar.html.erb
+++ b/plugins/redmine_tags/app/views/issues/_tags_sidebar.html.erb
@@ -1,4 +1,4 @@
-<% if defined?(sidebar_tags) && !sidebar_tags.empty? && !@issue %>
+<% if (defined?(sidebar_tags)) && @project.module_enabled?(:tags) && !sidebar_tags.empty? && !@issue %>
   <%= stylesheet_link_tag 'jquery.tagit.css', plugin: 'redmine_tags' %>
   <%= stylesheet_link_tag 'redmine_tags', plugin: 'redmine_tags' %>
   <h3><%= l :tags %></h3>
diff --git a/plugins/redmine_tags/app/views/wiki/_tags.html.erb b/plugins/redmine_tags/app/views/wiki/_tags.html.erb
index 3ee609a..58e21aa 100644
--- a/plugins/redmine_tags/app/views/wiki/_tags.html.erb
+++ b/plugins/redmine_tags/app/views/wiki/_tags.html.erb
@@ -1,4 +1,6 @@
+<% if @project.module_enabled?(:tags) -%>
 <% unless page.tag_list.empty? %>
   <b><%= l :tags %>:</b>
   <%= safe_join(page.tag_counts.collect{|t| render_tag_link(t, show_count: false, open_only: false) }, RedmineTags.settings[:issues_use_colors].to_i > 0 ? ' ' : ', ') %>
 <% end %>
+<% end %>
diff --git a/plugins/redmine_tags/app/views/wiki/_tags_form.html.erb b/plugins/redmine_tags/app/views/wiki/_tags_form.html.erb
index f92f59f..3acd01f 100644
--- a/plugins/redmine_tags/app/views/wiki/_tags_form.html.erb
+++ b/plugins/redmine_tags/app/views/wiki/_tags_form.html.erb
@@ -1,3 +1,4 @@
+<% if @project.module_enabled?(:tags) && User.current.allowed_to?(:wiki_edit_tags, @project) %>
 <p id="wiki_tags">
   <%= label_tag l(:tags), nil, for: :wiki_page_tag_list %>
   <%= text_field_tag 'wiki_page[tag_list]', page.tag_list.to_s, class: 'hol',
@@ -23,3 +24,4 @@
     caseSensitive: false,
     removeConfirmation: true,
   });" %>
+<% end %>
diff --git a/plugins/redmine_tags/app/views/wiki/_tags_sidebar.html.erb b/plugins/redmine_tags/app/views/wiki/_tags_sidebar.html.erb
index 834f154..3aa099a 100644
--- a/plugins/redmine_tags/app/views/wiki/_tags_sidebar.html.erb
+++ b/plugins/redmine_tags/app/views/wiki/_tags_sidebar.html.erb
@@ -1,6 +1,8 @@
+<% if @project.module_enabled?(:tags) -%>
 <% unless sidebar_tags.empty? %>
   <%= stylesheet_link_tag 'jquery.tagit.css', plugin: 'redmine_tags' %>
   <%= stylesheet_link_tag 'redmine_tags', plugin: 'redmine_tags' %>
   <h3><%= l :tags %></h3>
   <%= render_sidebar_tags %>
 <% end %>
+<% end %>
diff --git a/plugins/redmine_tags/init.rb b/plugins/redmine_tags/init.rb
index ac44cb0..fbf9813 100755
--- a/plugins/redmine_tags/init.rb
+++ b/plugins/redmine_tags/init.rb
@@ -17,6 +17,10 @@ Redmine::Plugin.register :redmine_tags do
 
   requires_redmine version_or_higher: '3.0.0'
 
+  project_module :tags do
+    permission :issue_edit_tags, { }
+    permission :wiki_edit_tags, { }
+  end
   settings \
     default:  {
       issues_sidebar:    'none',
@@ -26,6 +30,8 @@ Redmine::Plugin.register :redmine_tags do
       issues_sort_order: 'asc'
     },
     partial: 'tags/settings'
+
+
 end
 
 Rails.application.config.after_initialize do
